app/views/static/generic_object/generic_object_table.html.haml

```diff
@@ -5,15 +5,25 @@
   %tbody
     %tr{"ng-repeat" => "key in vm.keys track by $index"}
       %td.form-group{"ng-class" => "{'has-error': vm.angularForm['{{vm.keyType}}{{$index}}'].$invalid}"}
-        %input.form-control{"type"        => "text",
-                            "id"          => "{{vm.keyType}}{{$index}}",
-                            "name"        => "{{vm.keyType}}{{$index}}",
-                            "ng-required" => "vm.valueOptions && vm.values[$index] || {{vm.requiredRule}}",
-                            "ng-model"    => "vm.keys[$index]",
-                            "ng-change"   => "vm.tableCellValueChanged(vm.angularForm[vm.keyType+$index])",
-                            "ng-pattern"  => "/^[a-z][a-zA-Z_0-9]*[!?]?$/",
-                            "title"       => "{{vm.angularForm[vm.keyType+$index].$error.pattern ? vm.invalidKeyName : ''}}",
-                            "maxlength"   => "#{ViewHelper::MAX_NAME_LEN}",}
+        %div{"ng-if" => "vm.noPattern"}
+          %input.form-control{"type"        => "text",
+                              "id"          => "{{vm.keyType}}{{$index}}",
```

Comment:
Any chance this could be an array access instead? (`{{key.type}}` perhaps? Or at least `vm.keyTypes[$index]`?)

Don't use computed property names if you can avoid it, please :).